src/tools/pkcs11-tool.c

```diff
@@ -753,6 +758,19 @@ int main(int argc, char * argv[])
 		util_fatal("Cannot set FMODE to O_BINARY");
 #endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+		if (!osslctx) {
```

Comment:
we are in main. Is this check needed? I think in this case, it will be always true, unless we will introduce some wild goto logic. The same for the `default_provider` check below

Reply:
The tests could be removed if you want.